packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -242,6 +243,35 @@ exports.sourceNodes = async (
     )
   }
 
+  // Add referenced values
+  promises = []
+  for (let [key, references] of foreignReferenceMap) {
+    const node = getNode(key)
+    const nodeValues = {}
+    if (node) {
+      for (const field in references) {
+        nodeValues[field] = []
+        if (node[field]) {
+          nodeValues[field] = node[field]
+        }
+        references[field].forEach(reference => {
+          if (getNode(reference)) {
+            nodeValues[field].push(reference)
+          }
+        })
+      }
+
+      const newNode = {
+        ...node,
+        ...nodeValues,
+      }
+
+      newNode.internal.contentDigest = createContentDigest(stringify(newNode))
+      promises.push(createNode(newNode))
```

Comment:
this will be rewritte in my next PR so we only call createNode once instead of twice